modules/basics/src/main/java/com/opengamma/strata/basics/index/FloatingRateName.java

```diff
@@ -125,24 +126,25 @@ public static FloatingRateName of(String uniqueName) {
 
   // parse the config file FloatingRateName.ini
   private static ImmutableMap<String, FloatingRateName> parseIndices(IniFile ini) {
-    ImmutableMap.Builder<String, FloatingRateName> builder = ImmutableMap.builder();
-    PropertySet iborSection = ini.section("ibor");
-    for (String key : iborSection.keys()) {
-      builder.put(key, new FloatingRateName(key, iborSection.value(key) + "-", FloatingRateType.IBOR));
-    }
-    PropertySet onCompoundedSection = ini.section("overnightCompounded");
-    for (String key : onCompoundedSection.keys()) {
-      builder.put(key, new FloatingRateName(key, onCompoundedSection.value(key), FloatingRateType.OVERNIGHT_COMPOUNDED));
-    }
-    PropertySet onAveragedSection = ini.section("overnightAveraged");
-    for (String key : onAveragedSection.keys()) {
-      builder.put(key, new FloatingRateName(key, onAveragedSection.value(key), FloatingRateType.OVERNIGHT_AVERAGED));
-    }
-    PropertySet priceSection = ini.section("price");
-    for (String key : priceSection.keys()) {
-      builder.put(key, new FloatingRateName(key, priceSection.value(key), FloatingRateType.PRICE));
+    HashMap<String, FloatingRateName> mutableMap = new HashMap<>();
+    parseSection(ini.section("ibor"), "-", FloatingRateType.IBOR, mutableMap);
+    parseSection(ini.section("overnightCompounded"), "", FloatingRateType.OVERNIGHT_COMPOUNDED, mutableMap);
+    parseSection(ini.section("overnightAveraged"), "", FloatingRateType.OVERNIGHT_AVERAGED, mutableMap);
+    parseSection(ini.section("price"), "", FloatingRateType.PRICE, mutableMap);
+    return ImmutableMap.copyOf(mutableMap);
+  }
+
+  // parse a single section
+  private static void parseSection(
+      PropertySet section,
+      String indexNameSuffix,
+      FloatingRateType type,
+      HashMap<String, FloatingRateName> mutableMap) {
+
+    // find our names from the RHS of the key/value pairs
+    for (String key : section.keys()) {
+      mutableMap.put(key, new FloatingRateName(key, section.value(key) + indexNameSuffix, type));
```

Comment:
By replacing the map builder, we're losing the automatic check for duplicate keys - is this likely to be an issue? The last duplicate key seen will now win.